lua/entities/gmod_wire_coldetector.lua

```diff
@@ -0,0 +1,227 @@
+	
+	local posxd = ent:GetPhysicsObject():GetPos()
+	local anglesxd = ent:GetPhysicsObject():GetAngles()
+	if data.HitEntity == nil then return end
+	
+	self.Collided = true
+	self.CurrData = data
+	
+	WireLib.TriggerOutput(self, "Collided", 1)
+	WireLib.TriggerOutput(self, "Delta Time", data.DeltaTime)
+	WireLib.TriggerOutput(self, "Hit Entity", data.HitEntity)
+	WireLib.TriggerOutput(self, "Hit Position", data.HitPos)
+	WireLib.TriggerOutput(self, "Velocity", data.OurOldVelocity)
+	WireLib.TriggerOutput(self, "Hit Entity Velocity", data.TheirOldVelocity)
+	WireLib.TriggerOutput(self, "Speed", data.Speed)
+	WireLib.TriggerOutput(self, "Hit Normal", data.HitNormal)
+	WireLib.TriggerOutput(self, "Entity Position", posxd)
+	WireLib.TriggerOutput(self, "Entity Angle", anglesxd)
+	WireLib.TriggerOutput(self, "Hit Entity Position", ent2pos)
+	WireLib.TriggerOutput(self, "Hit Entity Angle", ent2ang)
+	
+	self:ShowOutput()
+	
+	if self.StickToWorld == 0 and self.StickToNPC == 0 and self.StickToProp == 0 then return end
+	
+	if GetConVarNumber("wire_col_detector_stickplayer") == 0 and data.HitEntity:IsPlayer() then return end
+	if GetConVarNumber("wire_col_detector_stickprops") == 0 and not gamemode.Call("CanProperty",self:GetPlayer(),"weld",data.HitEntity) then return end
+	
+	if self.StickToWorld == 0 and data.HitEntity:IsWorld() then return end
+	if self.StickToNPC == 0 and (data.HitEntity:IsNPC() or data.HitEntity:IsPlayer()) then return end
+	
+	local ent2pos = data.HitEntity:GetPos()
+	local ent2ang = data.HitEntity:GetAngles()
+	
+	timer.Simple(0,function()
+		if IsValid(ent) and (IsValid(data.HitEntity) or data.HitEntity:IsWorld()) and not IsValid(self.Stick) and  not IsValid(ent:GetParent()) then
+			local bonehit = 0
+			self.StickTo = data.HitEntity
+			
+			angle.pitch = ent2ang.pitch-angle.pitch
+			angle.yaw = ent2ang.yaw-angle.yaw
+			angle.roll = ent2ang.roll-angle.roll
+			vec:Rotate(angle)
+			
+			for i = 0,data.HitEntity:GetPhysicsObjectCount()-1 do
+				if data.HitEntity:GetPhysicsObjectNum(i) == data.HitObject then
+					bonehit = i
+					break
+				end
+			end
+			
+			local newentpos = data.HitEntity:GetPos()
+			local newentang = data.HitEntity:GetAngles()
+			
+			data.HitEntity:SetPos(ent2pos)
+			data.HitEntity:SetAngles(ent2ang)
+			ent:GetPhysicsObject():SetAngles(anglesxd)
+			ent:GetPhysicsObject():SetPos(posxd)
+			ent:SetAngles(anglesxd)
+			ent:SetPos(posxd)
+			
+			if util.IsValidPhysicsObject( data.HitEntity, 0) && (not self.NoPhysics || data.HitEntity:IsWorld()) then
+				self.Stick = constraint.Weld(ent, data.HitEntity,0,bonehit,0,true)
+				if data.HitEntity:IsWorld() then
+					ent:GetPhysicsObject():EnableMotion(false)
+					ent:GetPhysicsObject():Sleep()
+				end
+			else
+				ent:PhysicsDestroy()
+				ent:SetSolid(SOLID_NONE)
+				ent:FollowBone(data.HitEntity,bonehit)
+				
+			end
+			
+			data.HitEntity:SetPos(newentpos)
+			data.HitEntity:SetAngles(newentang)
```

Comment:
@Divran do you think these two can be moved before the if block?
If so, the if can be shortcut a bit.